File: ballast-core/dotnet/src/application/services/SignInService.cs
using Ballast.Core.Application.Models;
using System;
using System.Threading.Tasks;

namespace Ballast.Core.Application.Services
{
    public interface ISignInService : IDisposable
    {
        Task<PlayerDto> SignInAsync(PlayerSignInRequest request);
        Task SignOutAsync(PlayerSignOutRequest request);
        Task<PlayerDto> GetSignedInPlayerAsync(Guid playerId);
    }
}

File: ballast-core/npm/src/application/services/sign-in-service.ts
import { IDisposable } from "../../interfaces/disposable";
import { IPlayerDto } from "../models/player-dto";
import { IPlayerSignInRequest } from "../models/player-sign-in-request";
import { IPlayerSignOutRequest } from "../models/player-sign-out-request";

export interface ISignInService extends IDisposable {
    signInAsync(request: IPlayerSignInRequest): Promise<IPlayerDto>;
    signOutAsync(request: IPlayerSignOutRequest): Promise<void>;
    getSignedInPlayerAsync(playerId: string): Promise<IPlayerDto | null>;
}